src/util/configHelpers.js

```diff
@@ -722,6 +885,18 @@ const mergeListingConfig = (hostedConfig, defaultConfigs) => {
   };
 };
 
+const mergeUserConfig = (hostedConfig, defaultConfigs) => {
+  const hostedUserFields = restructureUserFields(hostedConfig?.userFields?.userFields);
+
+  const { userFields: defaultUserFields } = defaultConfigs.user;
+
+  // TODO: add shouldMerge logic
```

Comment:
Prepare this first part (config & show on profile page) as feature complete. I.e. avoid leaving TODOs behind.